src/components/reports/cost.ts

```diff
@@ -0,0 +1,169 @@
+import moment from 'moment';
+
+import { BillingClient } from '../../lib/billing';
+import CloudFoundryClient from '../../lib/cf';
+import { IOrganization, IOrganizationQuota } from '../../lib/cf/types';
+import { IParameters, IResponse } from '../../lib/router';
+
+import { IContext } from '../app/context';
+
+import costReportTemplate from './cost.njk';
+
+interface ICostable {
+  readonly incVAT: number;
+  readonly exVAT: number;
+}
+
+type ICostSummary = ICostable;
+
+interface IOrgCostRecord extends ICostable {
+  readonly orgGUID: string;
+  readonly orgName: string;
+
+  readonly quotaGUID: string;
+  readonly quotaName: string;
+}
+
+interface IQuotaCostRecord extends ICostable {
+  readonly quotaGUID: string;
+  readonly quotaName: string;
+}
+
+export async function viewCostReport(
+  ctx: IContext,
+  params: IParameters,
+): Promise<IResponse> {
+  const rangeStart = moment(params.rangeStart, 'YYYY-MM-DD').toDate();
+  const rangeStop  = moment(rangeStart).add(1, 'month').toDate();
+
+  const billingClient = new BillingClient({
+    apiEndpoint: ctx.app.billingAPI,
+    accessToken: ctx.token.accessToken,
+  });
+
+  const cf = new CloudFoundryClient({
+    accessToken: ctx.token.accessToken,
+    apiEndpoint: ctx.app.cloudFoundryAPI,
+  });
+
+  const orgs = await cf.organizations();
+  const orgGUIDs = orgs.map(o => o.metadata.guid);
+
+  const orgQuotas: {[key: string]: IOrganizationQuota} = (await Promise.all(
+    orgs.map(async (org: IOrganization) => {
+      const quota = await cf.organizationQuota(org.entity.quota_definition_guid);
```

Comment:
Now, this _probably_ won't ever be a problem to our tenants, but it could be to admin users. I'm wondering if `Promise.all()` would be a better thing to do, meaning the calls are parallel as oppose to asynchronous.

- Not a blocker

Reply:
Maybe I misunderstand, but we _are_ doing this in parallel.

The orgs.map is a collection of promises, and then we wait for them all to resolve.

Reply:
Yeah, it's right there in line 52... Never mind, ignore.